raiden/api/rest.py

```diff
@@ -190,6 +190,11 @@ def _serve_webui(self, file='index.html'):
             assert file
             web3 = self.flask_app.config.get('WEB3_ENDPOINT')
             if web3 and 'config.' in file and file.endswith('.json'):
+                host = request.headers.get('Host')
+                if any(h in web3 for h in ('localhost', '127.0.0.1')) and host:
+                    _, _port = split_endpoint(web3)
+                    _host, _ = split_endpoint(host)
+                    web3 = "http://{}:{}".format(_host, _port)
```

Comment:
I'm not sure this will be useful in practice. We can't just assume that the eth node will be available (and accept connections) on port 8545 of whatever the `Host` header points to.

Reply:
Yes, I agree it's not trustable, but it's the only information we have about the relation of the client and the server (in cases where client is not from localhost). It should cover one or other corner case, but the main part is the test-and-fallback on the web3 connection directly on webUI.

Reply:
Right, with the fallback it shouldn't hurt.